guides/source/active_storage_overview.md

```diff
@@ -687,6 +687,38 @@ previewable files. You can also call these methods directly.
 [`representable?`]: https://api.rubyonrails.org/classes/ActiveStorage/Blob/Representable.html#method-i-representable-3F
 [`representation`]: https://api.rubyonrails.org/classes/ActiveStorage/Blob/Representable.html#method-i-representation
 
+### Lazy vs Immediate Loading
+
+By default, Active Storage will represent files lazily. This code:
+
+```ruby
+image_tag file.representation(resize_to_limit: [100, 100])
+```
+
+Will generate an `<img>` tag with the `src` pointing to the [`ActiveStorage::Representations::RedirectController`][].
+The browser will make a request to that controller, which will return a `302` redirect to the file on the remote service (or in [proxy mode](#proxy-mode), return the file contents). Loading the file lazily allows features like [single use URLs](#public-access) to work without slowing down your initial page loads.
+
+This works fine for most cases.
+
+If you want to generate URLs for images immediately, you can call `.processed.url`:
+
+```ruby
+image_tag file.representation(resize_to_limit: [100, 100]).processed.url
+```
+
+The Active Storage variant tracker improves performance of this, by storing a record in the database if the requested representation has happened before. Thus, the above code will only make an API call to the remote service (eg. S3) once, and once a variant is stored, will use that. The variant tracker runs automatically, but can be disabled through `config.active_storage.track_variants`.
+
+If you're rendering lots of images on a page, the above example could result in N+1 queries loading all the variant records. Use the named scopes on [`ActiveStorage::Attachment`][] to avoid N+1s.
+
+```ruby
+user.avatars.with_all_variant_records.each do |file|
+  image_tag file.representation(resize_to_limit: [100, 100]).processed.url
```

Comment:
This will process all N files on the first access right?

Reply:
Yes.

Reply:
Should we maybe at a warning that it can make page slow?

Reply:
I guess maybe this isn't direct enough...

![image](https://user-images.githubusercontent.com/509837/122455829-71f47680-cf72-11eb-8dd1-03e637ce8ec4.png)

I could do followup.

Reply:
This also bypasses HTTP caching, as it will create a new temporary url each time it's called.
So you need to either cache the url or users will download the file every time they access the page.